File: myfidpass/Views/Settings/XStyleSideBar/XStyleSettingsSideBar.swift
//
//  XStyleSettingsSideBar.swift
//  myfidpass
//
//  Panneau latéral style X — profil commerçant + navigation.
//

import SwiftUI

struct XStyleSettingsSideBar: View {
    @EnvironmentObject private var authService: AuthService

    @Binding var isExpanded: Bool
    @Binding var path: NavigationPath

    var notificationIconURL: String?
    var hasNotificationIcon: Bool = false
    var onOpenFlyer: () -> Void
    var onOpenFootballGame: () -> Void
    var onOpenLiveGame: () -> Void

    @State private var showsFlyerCreationAttention = false

    private static let businessAvatarSize: CGFloat = 60

    private var displayName: String {
        if let slug = AuthStorage.currentBusinessSlug?.trimmingCharacters(in: .whitespacesAndNewlines),
           let biz = authService.businesses.first(where: { $0.slug == slug }) {
            let org = biz.organizationName?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
            if !org.isEmpty { return org }
            let n = biz.name.trimmingCharacters(in: .whitespacesAndNewlines)
            if !n.isEmpty { return n }
            return biz.slug
        }
        let email = (AuthStorage.userEmail ?? "").trimmingCharacters(in: .whitespacesAndNewlines)
        if !email.isEmpty { return email.components(separatedBy: "@").first ?? email }
        return "Mon commerce"
    }

    private var commerceCountLabel: String {
        "\(authService.businesses.count)"
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            profileAvatar
                .padding(.bottom, 10)

            Text(displayName)
                .font(.title3)
                .fontWeight(.semibold)
                .foregroundStyle(.white)
                .lineLimit(2)

            HStack(spacing: 2) {
                Text(commerceCountLabel)
                    .fontWeight(.bold)
                    .foregroundStyle(.white)
                Text(authService.businesses.count == 1 ? "commerce" : "commerces")
                    .foregroundStyle(.white.opacity(0.62))
            }
            .font(.callout)
            .fontWeight(.medium)
            .padding(.top, 5)

            ScrollView(.vertical) {
                VStack(alignment: .leading, spacing: 30) {
                    sideMenuButton(
                        icon: "doc.richtext",
                        title: "Flyer de jeu",
                        showsAttentionDot: showsFlyerCreationAttention,
                        dismissMenuOnTap: false
                    ) {
                        onOpenFlyer()
                    }

                    sideMenuButton(icon: "soccerball", title: "Jeu de foot", dismissMenuOnTap: false) {
                        onOpenFootballGame()
                    }

                    sideMenuButton(icon: "play.circle", title: "Jeu en direct", showsTrailingArrow: true, dismissMenuOnTap: false) {
                        onOpenLiveGame()
                    }

                    sideMenuButton(icon: "gearshape", title: "Paramètres") {
                        path.appendAnimated(SettingsSideRoute.settings)
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.top, 20)
            }
            .mask {
                Rectangle()
                    .ignoresSafeArea()
            }
            .overlay(alignment: .top) {
                Divider()
                    .background(.white.opacity(0.22))
                    .padding(.horizontal, -15)
            }
            .padding(.top, 15)
            .scrollClipDisabled()
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .leading)
        .padding([.horizontal, .top], 15)
        .background(Color.black)
        .onAppear { refreshFlyerCreationAttention() }
        .onReceive(NotificationCenter.default.publisher(for: .myfidpassMerchantSetupProgressUpdated)) { _ in
            refreshFlyerCreationAttention()
        }
        .onReceive(NotificationCenter.default.publisher(for: .myfidpassCardPreviewDisplayDidChange)) { _ in
            refreshFlyerCreationAttention()
        }
    }

    private func refreshFlyerCreationAttention() {
        let slug = AuthStorage.currentBusinessSlug?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        guard !slug.isEmpty else {
            showsFlyerCreationAttention = false
            return
        }
        CommerceFlyerStore.shared.hydrateFromDiskIfNeeded(slug: slug)
        showsFlyerCreationAttention = PostCardFlyerPromoEligibility.showsCreationAttentionBadge(for: slug)
    }

    @ViewBuilder
    private var profileAvatar: some View {
        sidebarBusinessAvatar(size: Self.businessAvatarSize)
    }

    @ViewBuilder
    private func sidebarBusinessAvatar(size: CGFloat) -> some View {
        let appIconShape = RoundedRectangle(cornerRadius: max(10, size * 0.28), style: .continuous)
        let icon = notificationIconURL?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        if hasNotificationIcon, !icon.isEmpty {
            BusinessLogoView(
                logoURL: icon,
                logoAssetContext: .campaignNotificationIcon,
                size: size,
                cornerRadius: max(10, size * 0.28)
            )
            .clipShape(appIconShape)
        } else {
            sidebarBusinessAvatarFallback(size: size)
        }
    }

    private func sidebarBusinessAvatarFallback(size: CGFloat) -> some View {
        RoundedRectangle(cornerRadius: max(10, size * 0.28), style: .continuous)
            .fill(
                LinearGradient(
                    colors: [Color(red: 0.05, green: 0.16, blue: 0.45), Color(red: 0.41, green: 0.12, blue: 0.55)],
                    startPoint: .topLeading,
                    endPoint: .bottomTrailing
                )
            )
            .overlay {
                Image(systemName: "storefront.fill")
                    .font(.system(size: max(13, size * 0.34), weight: .semibold))
                    .foregroundStyle(.white.opacity(0.95))
            }
            .frame(width: size, height: size)
    }

    @ViewBuilder
    private func sideMenuButton(
        icon: String,
        title: String,
        showsAttentionDot: Bool = false,
        showsTrailingArrow: Bool = false,
        dismissMenuOnTap: Bool = true,
        action: @escaping () -> Void
    ) -> some View {
        Button {
            if dismissMenuOnTap {
                isExpanded = false
            }
            action()
        } label: {
            HStack(spacing: 10) {
                Image(systemName: icon)
                    .font(.title3)
                    .frame(width: 30)
                    .symbolVariant(.fill)

                HStack(spacing: 8) {
                    Text(title)
                        .font(.title3)
                        .fontWeight(.bold)
                    if showsAttentionDot {
                        Circle()
                            .fill(Color.red)
                            .frame(width: 9, height: 9)
                            .accessibilityHidden(true)
                    }
                }

                if showsTrailingArrow {
                    Spacer(minLength: 0)
                    Image(systemName: "arrow.right")
                        .font(.body.weight(.bold))
                }
            }
            .foregroundStyle(.white)
        }
        .buttonStyle(.plain)
        .accessibilityLabel(
            showsAttentionDot ? "\(title), à créer" : title
        )
    }
}

File: android/app/src/main/java/fr/myfidpass/ui/components/XStyleSettingsSidebar.kt
package fr.myfidpass.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Article
import androidx.compose.material.icons.filled.NorthEast
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.SportsSoccer
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import fr.myfidpass.data.local.SessionStore

/** Contenu du panneau latéral — aligné iOS `XStyleSettingsSideBar`. */
@Composable
fun XStyleSettingsSidebar(
    sessionStore: SessionStore,
    onOpenFlyer: () -> Unit,
    onOpenFootballGame: () -> Unit,
    onOpenLiveGame: () -> Unit,
    onOpenSettings: () -> Unit,
    showFlyerAttentionDot: Boolean = false,
    modifier: Modifier = Modifier,
) {
    val businesses = sessionStore.businesses
    val currentSlug = sessionStore.currentBusinessSlug
    val current = businesses.firstOrNull { it.slug == currentSlug } ?: businesses.firstOrNull()
    val displayName = current?.name?.takeIf { it.isNotBlank() }
        ?: sessionStore.userEmail?.substringBefore("@")?.takeIf { it.isNotBlank() }
        ?: "Mon commerce"
    val avatarInitial = displayName.firstOrNull()?.uppercaseChar()?.toString() ?: "M"

    Column(
        modifier
            .fillMaxHeight()
            .background(Color.Black)
            .padding(horizontal = 15.dp, vertical = 15.dp),
    ) {
        Box(
            Modifier
                .size(60.dp)
                .clip(RoundedCornerShape(17.dp))
                .background(
                    Brush.linearGradient(listOf(Color(0xFF0D2A73), Color(0xFF691E8C))),
                ),
            contentAlignment = Alignment.Center,
        ) {
            Text(avatarInitial, color = Color.White, fontWeight = FontWeight.Bold, fontSize = 24.sp)
        }
        Spacer(Modifier.height(10.dp))
        Text(
            displayName,
            color = Color.White,
            fontWeight = FontWeight.SemiBold,
            fontSize = 20.sp,
            lineHeight = 24.sp,
        )
        Row(Modifier.padding(top = 5.dp)) {
            Text(
                "${businesses.size}",
                color = Color.White,
                fontWeight = FontWeight.Bold,
                fontSize = 15.sp,
            )
            Text(
                if (businesses.size == 1) " commerce" else " commerces",
                color = Color.White.copy(0.62f),
                fontSize = 15.sp,
            )
        }
        HorizontalDivider(
            Modifier.padding(top = 15.dp),
            color = Color.White.copy(0.22f),
        )
        Column(
            Modifier
                .weight(1f)
                .verticalScroll(rememberScrollState())
                .padding(top = 20.dp),
            verticalArrangement = Arrangement.spacedBy(30.dp),
        ) {
            SidebarMenuButton(
                icon = { Icon(Icons.AutoMirrored.Filled.Article, contentDescription = null, tint = Color.White) },
                title = "Flyer de jeu",
                showAttentionDot = showFlyerAttentionDot,
                onClick = onOpenFlyer,
            )
            SidebarMenuButton(
                icon = { Icon(Icons.Default.SportsSoccer, contentDescription = null, tint = Color.White) },
                title = "Jeu de foot",
                onClick = onOpenFootballGame,
            )
            SidebarMenuButton(
                title = "Jeu en direct",
                showInlineExternalArrow = true,
                onClick = onOpenLiveGame,
            )
            SidebarMenuButton(
                icon = { Icon(Icons.Default.Settings, contentDescription = null, tint = Color.White) },
                title = "Paramètres",
                onClick = onOpenSettings,
            )
        }
    }
}

@Composable
private fun SidebarMenuButton(
    title: String,
    onClick: () -> Unit,
    icon: (@Composable () -> Unit)? = null,
    showAttentionDot: Boolean = false,
    showInlineExternalArrow: Boolean = false,
) {
    Row(
        Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(14.dp),
    ) {
        if (icon != null) {
            Box(Modifier.size(24.dp), contentAlignment = Alignment.Center) {
                icon()
            }
        }
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(if (showInlineExternalArrow) 5.dp else 8.dp),
        ) {
            Text(
                title,
                color = Color.White,
                fontWeight = FontWeight.Medium,
                fontSize = 17.sp,
            )
            if (showInlineExternalArrow) {
                Icon(
                    Icons.Default.NorthEast,
                    contentDescription = null,
                    tint = Color.White,
                    modifier = Modifier.size(16.dp),
                )
            }
            if (showAttentionDot) {
                Box(
                    Modifier
                        .size(8.dp)
                        .clip(RoundedCornerShape(999.dp))
                        .background(Color(0xFFEF4444)),
                )
            }
        }
    }
}
